src/System.Private.CoreLib/src/System/Runtime/Intrinsics/X86/Sse42.PlatformNotSupported.cs

```diff
@@ -18,259 +18,386 @@ public static class Sse42
 
         /// <summary>
         /// int _mm_cmpestri (__m128i a, int la, __m128i b, int lb, const int imm8)
         ///   PCMPESTRI xmm, xmm/m128, imm8
         /// </summary>
-        public static int CompareExplicitLengthIndex(Vector128<sbyte> left, byte leftLength, Vector128<sbyte> right, byte rightLength, StringComparisonMode mode) { throw new PlatformNotSupportedException(); }
+        public static int CompareIndex(Vector128<sbyte> left, int leftLength, Vector128<sbyte> right, int rightLength, StringComparisonMode mode) { throw new PlatformNotSupportedException(); }
 
         /// <summary>
         /// int _mm_cmpestri (__m128i a, int la, __m128i b, int lb, const int imm8)
         ///   PCMPESTRI xmm, xmm/m128, imm8
         /// </summary>
-        public static int CompareExplicitLengthIndex(Vector128<byte> left, byte leftLength, Vector128<byte> right, byte rightLength, StringComparisonMode mode) { throw new PlatformNotSupportedException(); }
+        public static int CompareIndex(Vector128<byte> left, int leftLength, Vector128<byte> right, int rightLength, StringComparisonMode mode) { throw new PlatformNotSupportedException(); }
 
         /// <summary>
         /// int _mm_cmpestri (__m128i a, int la, __m128i b, int lb, const int imm8)
         ///   PCMPESTRI xmm, xmm/m128, imm8
         /// </summary>
-        public static int CompareExplicitLengthIndex(Vector128<short> left, byte leftLength, Vector128<short> right, byte rightLength, StringComparisonMode mode) { throw new PlatformNotSupportedException(); }
+        public static int CompareIndex(Vector128<short> left, int leftLength, Vector128<short> right, int rightLength, StringComparisonMode mode) { throw new PlatformNotSupportedException(); }
 
         /// <summary>
         /// int _mm_cmpestri (__m128i a, int la, __m128i b, int lb, const int imm8)
         ///   PCMPESTRI xmm, xmm/m128, imm8
         /// </summary>
-        public static int CompareExplicitLengthIndex(Vector128<ushort> left, byte leftLength, Vector128<ushort> right, byte rightLength, StringComparisonMode mode) { throw new PlatformNotSupportedException(); }
+        public static int CompareIndex(Vector128<ushort> left, int leftLength, Vector128<ushort> right, int rightLength, StringComparisonMode mode) { throw new PlatformNotSupportedException(); }
 
         /// <summary>
         /// __m128i _mm_cmpistrm (__m128i a, __m128i b, const int imm8)
         ///   PCMPISTRM xmm, xmm/m128, imm8
         /// </summary>
-        public static Vector128<ushort> CompareImplicitLengthBitMask(Vector128<sbyte> left, Vector128<sbyte> right, StringComparisonMode mode) { throw new PlatformNotSupportedException(); }
+        public static Vector128<ushort> CompareBitMask(Vector128<sbyte> left, Vector128<sbyte> right, StringComparisonMode mode) { throw new PlatformNotSupportedException(); }
```

Comment:
For my information - Why did we break out `BitMask` vs `UnitMask` into different overloads?  Why not keep it in the `StringComparisonMode` enum?

Reply:
Ah, good catch! This is a legacy design from the previous versions, will fix. Thanks!